analyses/survival-analysis/survival-analysis_subtypes.Rmd

```diff
@@ -99,40 +110,44 @@ This way we subset the metadata to 1 single primary sample per patient ID
 independent_primary <- readr::read_tsv(independent_primary_file)
 
 # select only independent primary bs_ids to get overall survival
-metadata <- metadata %>%
+metadata_ind_dna <- metadata %>%
   dplyr::filter(Kids_First_Biospecimen_ID %in%
-                  independent_primary$Kids_First_Biospecimen_ID)
+                  independent_primary$Kids_First_Biospecimen_ID) 
+
+# check whether any RNA pts not in DNA subset
+metadata_rna <- metadata %>%
+  filter(experimental_strategy == "RNA-Seq" & 
+           tumor_descriptor == "Initial CNS Tumor" & 
+           !is.na(OS_years)) %>%
+  unique()
+
+rna_pts_to_add <- setdiff(unique(metadata_rna$Kids_First_Participant_ID), metadata_ind_dna$Kids_First_Participant_ID)
+# nope
+
 
 # find molecular subtypes which have more than n_participants (3) participants
-filter_molecular_subtype <- metadata %>% group_by(molecular_subtype) %>% dplyr::tally() %>% dplyr::filter(n>n_participants)
+filter_molecular_subtype <- metadata_ind_dna %>% group_by(molecular_subtype) %>% dplyr::tally() %>% dplyr::filter(n>n_participants)
 
-# select only molecular_subtypes that have more than n_participants (3) participants
-metadata <- metadata %>%
+# select only  that have more than n_participants (3) participants
+metadata_ind_dna <- metadata_ind_dna %>%
   dplyr::filter(molecular_subtype %in%
                   filter_molecular_subtype$molecular_subtype)
```

Comment:
This can be simplified to only 1 line of code:

```
metadata_ind_dna <- metadata_ind_dna %>%
  group_by(molecular_subtype) %>% 
  # directly add a column for the counts instead of tallying down into a summarized tibble
  mutate(n_molecular_subtype = n()) %>%
  # keep only sufficient n
  filter(n_molecular_subtype>n_participants) %>%
  # rm the counting column and it's all cleaned up
  select(-n_molecular_subtype)
```

Reply:
We still have two groups going into modeling that only have 1 sample each: `HGG, IDH, TP53 activated ` and `HGG, H3 G35`. That means there is no variation in the groups to model. Hence, we see them coming up with _both_ CI's as `NA` and an unreliable estimate.

On one hand I would not want to include groupings that have insufficient data, but on the other hand if the goal is to compare among groups we cannot cherry-pick them. My personal assessment here is that there is not sufficient data to rely on the model output.

Reply:
I was wondering why we didn't see the G35 in the final table and it is because there aren't enough samples per the n=3 per subtype rule set in place in this notebook prior to my edits. I think some are also not primary tumors, so I'll check why they're even in the model..?

Reply:
I am not seeing these in the model either